### PR TITLE
Support legacy plugins

### PR DESCRIPTION
**Pull request by smithsz**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/nodejs-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.md](https://github.com/cloudant/nodejs-cloudant/blob/master/CHANGES.md)
- [x] You have completed the PR template below:

## What
Support legacy plugins.

## How
Reinstate old code to enable users to use custom legacy plugins.

## Testing
Includes unit tests.

### CODE REVIEW

File: cloudant.js [hunk 1]
```diff
@@ -47,15 +50,51 @@ function Cloudant(options, callback) {
...
+    debug('Using legacy custom plugin');
+    cloudantRequest = options.plugin;
+
+    if (options.requestDefaults) {
+      requestDefaults = options.requestDefaults;
+    } else {
+      requestDefaults = { gzip: true, jar: false };
+    }
+
+    // set library UA header
```
Comment (ricellis) [inline cloudant.js]: Do we have two copies of this now?

Comment (smithsz) [reply cloudant.js]: Yeah, this is a legacy code path. If the client specifies a legacy plugin then we avoid initialising the Cloudant client and instead substitute in their plugin. We still need to add the UA and setup the default agent config.

Comment (ricellis) [reply cloudant.js]: It'd be nice if we could at least use the same agent/constants in both cases, but if it's not easy to put the code somewhere common then I'm ok with this.

**Comment by smithsz** [comment]
I think we can pass a legacy plugin to the new `CloudantClient` where it'll be used as a direct replacement of `request`. This means we can support the use of new plugins alongside one legacy plugin. New PR to follow.

**Pull request closed without merging**